pkg/csi/service/common/util.go

```diff
@@ -242,16 +241,18 @@ func ParseStorageClassParams(ctx context.Context, params map[string]string) (*St
 				param = strings.ToLower(param)
 				if param == DatastoreMigrationParam {
 					scParams.Datastore = value
-				} else if param != DiskFormatMigrationParam && param != HostFailuresToTolerateMigrationParam &&
-					param != ForceProvisioningMigrationParam && param != CacheReservationMigrationParam &&
-					param != DiskstripesMigrationParam && param != ObjectspacereservationMigrationParam &&
-					param != IopslimitMigrationParam {
-					return nil, fmt.Errorf("Invalid parameter key:%v, value:%v", param, value)
+				} else if param == DiskFormatMigrationParam || param == HostFailuresToTolerateMigrationParam ||
+					param == ForceProvisioningMigrationParam || param == CacheReservationMigrationParam ||
+					param == DiskstripesMigrationParam || param == ObjectspacereservationMigrationParam ||
+					param == IopslimitMigrationParam {
+					return nil, fmt.Errorf("vSphere CSI driver does not support creating volume using in-tree vSphere volume plugin parameter key:%v, value:%v", param, value)
+				} else {
+					return nil, fmt.Errorf("invalid parameter. key:%v, value:%v", param, value)
```

Comment:
Which params fall under this category?

Reply:
These are parameters which does not fall under Valid Migration Parameter or Valid CSI driver parameters.
For example take a look at following storageclass.

```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: example-vanilla-block-sc
provisioner: csi.vsphere.vmware.com
parameters:
  csimigration: "true"
  test1: "value1"
```

We should see following error message.

> 2020-09-15T16:34:55.201609894Z 2020-09-15T16:34:55.201Z	INFO	vanilla/controller.go:494	CreateVolume: called with args {Name:pvc-5c0b7997-1c9e-482d-bfc7-e12ae9dccb41 CapacityRange:required_bytes:5368709120  VolumeCapabilities:[mount:<> access_mode:<mode:SINGLE_NODE_WRITER > ] Parameters:map[test1:value1] Secrets:map[] VolumeContentSource:<nil> AccessibilityRequirements:<nil> XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}	{"TraceId": "646d8260-ca92-45d7-a7aa-eaa04fb30651"}
2020-09-15T16:34:55.202251013Z 2020-09-15T16:34:55.201Z	ERROR	vanilla/controller.go:280	Parsing storage class parameters failed with error: invalid parameters :map[test1:value1]	{"TraceId": "646d8260-ca92-45d7-a7aa-eaa04fb30651"}